src/nbl/ext/ImGui/ImGui.cpp

```diff
@@ -1235,32 +1482,33 @@ namespace nbl::ext::imgui
 				draw_data->DisplayPos+data_data->DisplaySize (bottom right). DisplayPos is (0,0) for single viewport apps.
 			*/
 
+			const auto& [structureOffsets, limits] = std::make_tuple(imCmdRange.getRequiredStructsBlockInfo().offsets, imCmdRange.getLimits());
 			{
 				PushConstants constants
 				{
-					.elementBDA = { mdiBuffer->getDeviceAddress() + mdiOffsets[MDI::EBC_ELEMENT_STRUCTURES]},
-					.elementCount = { mdiParams.drawCount },
+					.elementBDA = { mdiBuffer->getDeviceAddress() + structureOffsets[ImGuiCommandListRange::STightStructs::ELEMENT_STRUCTURES] },
+					.elementCount = { limits.totalIndirectDrawCount },
 					.scale = { trs.scale[0u], trs.scale[1u] },
 					.translate = { trs.translate[0u], trs.translate[1u] },
 					.viewport = { viewport.x, viewport.y, viewport.width, viewport.height }
 				};
 
-				commandBuffer->pushConstants(pipeline->getLayout(), IShader::E_SHADER_STAGE::ESS_VERTEX | IShader::E_SHADER_STAGE::ESS_FRAGMENT, 0u, sizeof(constants), &constants);
+				commandBuffer->pushConstants(m_pipeline->getLayout(), IShader::E_SHADER_STAGE::ESS_VERTEX | IShader::E_SHADER_STAGE::ESS_FRAGMENT, 0u, sizeof(constants), &constants);
 			}
 
-			const asset::SBufferBinding<const video::IGPUBuffer> binding =
+			const SBufferBinding<const IGPUBuffer> binding =
 			{
-				.offset = mdiOffsets[MDI::EBC_DRAW_INDIRECT_STRUCTURES],
-				.buffer = core::smart_refctd_ptr(mdiBuffer)
+				.offset = structureOffsets[ImGuiCommandListRange::STightStructs::INDIRECT_STRUCTURES],
+				.buffer = smart_refctd_ptr(mdiBuffer)
 			};
 
-			commandBuffer->drawIndexedIndirect(binding, mdiParams.drawCount, sizeof(VkDrawIndexedIndirectCommand));
+			commandBuffer->drawIndexedIndirect(binding, limits.totalIndirectDrawCount, sizeof(VkDrawIndexedIndirectCommand));
```

Comment:
being able to split up your allocations into multiple blocks, requires that you can also split up your MDI allocation, therefore multiple calls to `drawIndexedIndirect` with different offset for your MDI buffer

Reply:
And if you're gonna do multiple calls to `drawIndexedIndirect` you must hoist the viewport, scissor setting and vertex attribute binding

Reply:
done by myself